components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/api/ReCaptchaApi.java

```diff
@@ -0,0 +1,103 @@
+ *
+ */
+
+package org.wso2.carbon.identity.mgt.endpoint.client.api;
+
+import com.sun.jersey.api.client.GenericType;
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.base.MultitenantConstants;
+import org.wso2.carbon.identity.mgt.endpoint.IdentityManagementEndpointConstants;
+import org.wso2.carbon.identity.mgt.endpoint.IdentityManagementServiceUtil;
+import org.wso2.carbon.identity.mgt.endpoint.client.ApiClient;
+import org.wso2.carbon.identity.mgt.endpoint.client.ApiException;
+import org.wso2.carbon.identity.mgt.endpoint.client.Configuration;
+import org.wso2.carbon.identity.mgt.endpoint.client.Pair;
+import org.wso2.carbon.identity.mgt.endpoint.client.model.ReCaptchaProperties;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class generating and invoking the Captcha api to display and the Recaptcha.
+ */
+public class ReCaptchaApi {
+
+    private final String[] localVarAccepts = {"application/json"};
+    private final String[] localVarContentTypes = {"application/json"};
+    private String basePath = IdentityManagementServiceUtil.getInstance().getServiceContextURL()
+            .replace(IdentityManagementEndpointConstants.UserInfoRecovery.SERVICE_CONTEXT_URL_DOMAIN,
+                    "api/identity/recovery/v0.9");
+    private ApiClient apiClient;
+
+    public ReCaptchaApi() {
+
+        this(Configuration.getDefaultApiClient());
+    }
+
+    private ReCaptchaApi(ApiClient apiClient) {
+
+        this.apiClient = apiClient;
+    }
+
+    /**
+     * return the reCaptchaGet details in the headers for the given tenant.
+     *
+     * @param tenantDomain tenant domain. Default &#x60;carbon.super&#x60; (optional)
+     * @throws ApiException if fails to make API call
+     */
+    public ReCaptchaProperties getReCaptcha(String tenantDomain, boolean isEndpointTenantAware, String captchaType,
+                                            String recoveryType) throws ApiException {
+
+        if (StringUtils.isBlank(tenantDomain)) {
+            tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+
+        if (isEndpointTenantAware && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equalsIgnoreCase(tenantDomain)) {
+            basePath = IdentityManagementServiceUtil.getInstance().getServiceContextURL()
+                    .replace(IdentityManagementEndpointConstants.UserInfoRecovery.SERVICE_CONTEXT_URL_DOMAIN,
+                            "t/" + tenantDomain + "/api/identity/recovery/v0.9");
+        }
+
+        apiClient.setBasePath(basePath);
+
+        String localVarPath = "/captcha";
+
+        // Query params
+        List<Pair> localVarQueryParams = new ArrayList<>();
+        Map<String, String> localVarHeaderParams = new HashMap<>();
+        Map<String, Object> localVarFormParams = new HashMap<>();
+
+        localVarQueryParams.addAll(apiClient.parameterToPairs(StringUtils.EMPTY, "tenant-domain", tenantDomain));
```

Comment:
can we use constants for the parameter keys since it's used in many places

Reply:
fixed